lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -369,82 +369,242 @@ void LCIStrategicPlugin::handleGreenSignalScenario(const cav_srvs::PlanManeuvers
+  traj_lower.is_algorithm_successful = false;
+  traj_lower.case_num = CASE_1;
+
+  traj_lower.t1_ = current_time + modified_remaining_time_lower;
+  traj_lower.v1_ = modified_departure_speed_lower;
+  traj_lower.a1_ = max_comfort_decel_;
+  traj_lower.x1_ = traffic_light_downtrack;
+
+  traj_lower.t2_ = traj_lower.t1_;
+  traj_lower.v2_ = traj_lower.v1_;
+  traj_lower.a2_ = traj_lower.a1_;
+  traj_lower.x2_ = traj_lower.x1_;
+  
+  traj_lower.t3_ = traj_lower.t2_;
+  traj_lower.v3_ = traj_lower.v2_;
+  traj_lower.a3_ = traj_lower.a2_;
+  traj_lower.x3_ = traj_lower.x2_;
 
-  // handle hard failure case such as nan
-  if (!isnan(params.modified_departure_speed) && params.modified_departure_speed > epsilon_ &&
-      params.modified_departure_speed <  speed_limit ) //80_mph
+  // Pick UPPER or LOWER trajectory based on light
+  bool is_return_params_found = false;
+  
+  if (calculation_success_upper)
+  {
+    auto upper_optional = traffic_light->predictState(lanelet::time::timeFromSec(current_time + modified_remaining_time_upper));
+
+    if (!validLightState(upper_optional, ros::Time(current_time + modified_remaining_time_upper)))
+    {
+      ROS_ERROR_STREAM("Unable to resolve give signal for modified_remaining_time_upper: " << std::to_string(current_time + modified_remaining_time_upper));
+    }
+    else
+    {
+      if (upper_optional.get().second == lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED)
+      {
+        ROS_DEBUG_STREAM("Detected Upper GREEN case");    
+        return_params = traj_upper;
+        is_return_params_found = true;
+      }
+    }
+  }
+ 
+  if (!is_return_params_found)
   {
-    ROS_DEBUG_STREAM("Updated the speed, and using modified_departure_speed: " << params.modified_departure_speed);
-    print_params(params);
-    return params;
+    auto lower_optional = traffic_light->predictState(lanelet::time::timeFromSec(current_time + modified_remaining_time_lower));
+
+    if (!validLightState(lower_optional, ros::Time(current_time + modified_remaining_time_lower)))
+    {
+      ROS_ERROR_STREAM("Unable to resolve give signal for modified_remaining_time_lower:" << std::to_string(current_time + modified_remaining_time_lower));
+    }
+    else
+    {
+      if (lower_optional.get().second == lanelet::CarmaTrafficSignalState::PROTECTED_MOVEMENT_ALLOWED)
+      {
+        ROS_DEBUG_STREAM("Detected Lower GREEN case");    
+        return_params = traj_lower;
+        is_return_params_found = true;
+      }
+    }
+  }
+
+  // Handle hard failure case such as nan or invalid states
+  if (is_return_params_found && !isnan(return_params.modified_departure_speed) && return_params.modified_departure_speed > epsilon_ &&
+      return_params.modified_departure_speed < speed_limit ) //80_mph
+  {
+    ROS_DEBUG_STREAM("Updated the speed, and using modified_departure_speed: " << return_params.modified_departure_speed);
+    print_params(return_params);
+    return return_params;
   }
```

Comment:
Let add logging to else statement to know if is_return_params_found is green and the speed is wrong.